File: java/src/binary_search/randomTestData.java
package binary_search;

import java.util.Arrays;
import java.util.Random;

class randomTestData {
	private static final Random rand = new Random(0);

	public int[] array;
	public int val;
	public int expectedFirst, expectedLast, expectedCount;

	randomTestData() {
		int n = rand.nextInt(50), max = rand.nextInt(50);
		array = new int[n];
		for (int i = 0; i < n; i++) {
			array[i] = rand.nextInt(1 + max);
		}
		val = rand.nextInt(1 + max);
		expectedFirst = -1;
		expectedLast = -1;

		Arrays.sort(array);
		for (int i = 0; i < n; i++) {
			if (array[i] == val) {
				if (expectedFirst < 0) {
					expectedFirst = i;
				}
				expectedLast = i;
				expectedCount++;
			}
		}
	}
}

File: go/binary_search/random_test_data.go
package binary_search

import (
	"math/rand/v2"
	"sort"
)

type randomTestData struct {
	array                                           []int
	val, expectedFirst, expectedLast, expectedCount int
}

var random = rand.New(rand.NewPCG(0, 0))

func newRandomTestData() *randomTestData {
	var td randomTestData

	n, maxVal := random.IntN(50), random.IntN(50)
	td.array = make([]int, n)
	for i := range n {
		td.array[i] = random.IntN(1 + maxVal)
	}

	td.val = random.IntN(1 + maxVal)
	td.expectedFirst = -1
	td.expectedLast = -1

	sort.Ints(td.array)
	for i := range n {
		if td.array[i] == td.val {
			if td.expectedFirst < 0 {
				td.expectedFirst = i
			}
			td.expectedLast = i
			td.expectedCount++
		}
	}

	return &td
}
